wps-scripts/src/main/resources/archetype-resources/src/main/java/WpsScriptsPackage.java

```diff
@@ -0,0 +1,316 @@
+        }
+        */
+    }
+
+    /**
+     * This method load the script contained in the 'scripts' folder of the resources an put the under the given a
+     * node with the given name. (Be careful before any modification)
+     *
+     * @param nodePath Name of the node containing the loaded scripts.
+     */
+    private void defaultLoadScript(String nodePath, String[] iconArrayName){
+        //Retrieve the 'scripts' folder path.
+        String folderPath = this.getClass().getResource("scripts").getFile();
+        //Load the icon and get back the corresponding array with file path (in the same order)
+        String[] array = loadIcons(iconArrayName);
+        //Recursively add the scripts.
+        recursiveScriptLoad(nodePath, folderPath, array);
+
+    }
+
+    private void recursiveScriptLoad(String nodePath, String folderPath, String[] iconArrayName){
+        //Get the URL of all the files contained in the 'script' folder.
+        Enumeration<URL> enumUrl = FrameworkUtil.getBundle(this.getClass()).findEntries(folderPath, "*", false);
+        //For each url, if it is a file, load it, if it is a directory, check its content.
+        while(enumUrl.hasMoreElements()){
+            try {
+                //Get the URL
+                URL scriptUrl = enumUrl.nextElement();
+                //If the url if a groovy file,
+                if(scriptUrl.getFile().endsWith(".groovy")) {
+                    //Create a temporary File object
+                    final File tempFile = File.createTempFile("wpsprocess", ".groovy");
+                    tempFile.deleteOnExit();
+                    //Copy the content of the resource file in the temporary file.
+                    try (FileOutputStream out = new FileOutputStream(tempFile)) {
+                        IOUtils.copy(scriptUrl.openStream(), out);
+                    }
+                    //Add the temporary file to the WpsService an get back the ProcessIdentifier object
+                    ProcessIdentifier pi = localWpsService.addLocalScript(tempFile, iconArrayName, true, nodePath);
+                    //Save the process id to be able to remove the process later.
+                    listIdProcess.add(pi.getProcess().getIdentifier());
+                }
+                //If the url is a folder,
+                else if(!scriptUrl.getFile().contains(".")){
+                    String folderName = new File(scriptUrl.getFile()).getName();
+                    //Recursively add the scripts.
+                    recursiveScriptLoad(nodePath+"/"+folderName, folderPath+File.separator+folderName, iconArrayName);
+                }
+            } catch (IOException e) {
+                LoggerFactory.getLogger(WpsScriptsPackage.class).error(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * This method loads the scripts one by one under different node path with different icons.
+     * (Be careful before any modification)
```

Comment:
In the archetype, there is only one script (to keep it the simplest), but the plugin can contains as many script as wanted.
The 'defaultLoadScript()' method uses the 'recursiveScriptLoad()' method to load all the scripts keeping the file tree structure.
The 'customLoadScript' should load the script one by one. As there is only one script in the archetype, only one script is loaded. If the user put more scripts, he should duplicate the code loading the signle script.
The user should use one of these two methods.